receive.c

```diff
@@ -392,31 +468,60 @@ int main(int argc, char **argv) {
 	}
 
 	if(filteropt == 1) {
-		struct protocol_t *protocol = NULL;
-		nrfilter = explode(filter, ",", &filters);
-		unsigned int match = 0, j = 0;
-
-		protocol_init();
-
-		for(j=0;j<nrfilter;j++) {
-			match = 0;
-			struct protocols_t *pnode = protocols;
-			if(filters[j] != NULL && strlen(filters[j]) > 0) {
-				while(pnode) {
-					protocol = pnode->listener;
-					if(protocol_device_exists(protocol, filters[j]) == 0 && match == 0) {
-						match = 1;
-						break;
+		if(incfilter != NULL) {
+			struct protocol_t *protocol = NULL;
+			nrincfilter = explode(incfilter, ",", &incfilters);
+			unsigned int match = 0, j = 0;
+
+			protocol_init();
```

Comment:
True, I'll move this and `struct protocol_t *protocol = NULL;` and `unsigned int match = 0, j = 0;` to just after `if(filteropt == 1) {`